### PR TITLE
AutoX feature

### PR DESCRIPTION
**Pull request by Alan-Cha**
Replace #1355

### CODE REVIEW

File: autox/informer.go [hunk 1]
```diff
@@ -3,154 +3,444 @@ package autox
...
+//	the name of the release group spec (releaseGroupSpecName)
+//	the ID of the release spec (releaseSpecID)
+func getReleaseName(releaseGroupSpecName string, releaseSpecID string) string {
+	return fmt.Sprintf("autox-%s-%s", releaseGroupSpecName, releaseSpecID)
 }
 
-func deleteHelmReleases(resourceName string) {
-	doChartAction(resourceName, deleteAction)
+// shouldCreateApplication will return true if an application should be created
+// an application should be created if the values are different from those from the previous application (if one exists)
```
Comment (sriumcp) [inline autox/informer.go]: // comment: Or if there is no pre-existing application.

File: autox/informer.go [hunk 2]
```diff
@@ -3,154 +3,444 @@ package autox
...
+	// check if name matches trigger
+	name := u.GetName()
+	if name != releaseGroupSpec.Trigger.Name {
 		return
 	}
 
-	// if the pruned labels are different, then update by deleting and reinstalling
-	deleteHelmReleases(resourceName)
-	installHelmReleases(resourceName)
+	// at this point, we know that we are really handling an even for the trigger object
```
Comment (sriumcp) [inline autox/informer.go]: `even(t)` -- spelling.

**Review by kalantar — APPROVED** [review]
no additional changes to recommend

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**